webapp/src/utils/translationObjectUtil.spec.ts

```diff
@@ -32,4 +37,73 @@ describe('translationObjectUtil', () => {
       expect(actual).toEqual(expected);
     });
   });
+  describe('getTranslationsBySourceFile()', () => {
+    it('returns empty object for empty obj', () => {
+      const actual = getTranslationsBySourceFile({});
+      expect(actual).toEqual({});
+    });
+    it('group by sourceFile obj one property', () => {
+      const actual = getTranslationsBySourceFile({
+        a: { sourceFile: 'en.yaml', text: 'A' },
+      });
+      expect(actual).toEqual({ 'en.yaml': { a: 'A' } });
+    });
+    it('group by sourceFile simple obj one object property', () => {
+      const actual = getTranslationsBySourceFile({
+        'a.b.c': { sourceFile: 'en.yaml', text: 'ABC' },
+      });
+      expect(actual).toEqual({ 'en.yaml': { 'a.b.c': 'ABC' } });
+    });
+    it('group by sourceFile two properties obj', () => {
+      const actual = getTranslationsBySourceFile({
+        'f1.a.b.c': { sourceFile: 'f1/en.yaml', text: 'ABC' },
+        'f2.a.b.e': { sourceFile: 'f2/en.yml', text: 'ABE' },
+      });
+      const expected = {
+        'f1/en.yaml': { 'a.b.c': 'ABC' },
+        'f2/en.yml': { 'a.b.e': 'ABE' },
+      };
+      expect(actual).toEqual(expected);
+    });
+    it('group by sourceFile two sourceFiles obj', () => {
+      const actual = getTranslationsBySourceFile({
+        'f1.a.b.c': { sourceFile: 'f1/en.yaml', text: 'ABC' },
+        'f1.a.b.e': { sourceFile: 'f1/en.yaml', text: 'ABE' },
+        'f2.a.b.f': { sourceFile: 'f2/en.yml', text: 'ABF' },
+      });
+      const expected = {
+        'f1/en.yaml': { 'a.b.c': 'ABC', 'a.b.e': 'ABE' },
+        'f2/en.yml': { 'a.b.f': 'ABF' },
+      };
+      expect(actual).toEqual(expected);
+    });
+  });
+  describe('getPrefixKeyFromSourceFile()', () => {
+    it('returns empty string for empty string', () => {
+      const actual = getPrefixKeyFromSourceFile('');
+      expect(actual).toEqual('');
+    });
+    it('returns empty string for no path', () => {
+      const actual = getPrefixKeyFromSourceFile('en.yaml');
+      expect(actual).toEqual('');
+    });
+    it('returns empty string for no path', () => {
+      const actual = getPrefixKeyFromSourceFile('sub1/sub2/en.yaml');
+      expect(actual).toEqual('sub1.sub2.');
+    });
```

Comment:
This test case copies the name of the test case above, but tests the a case with a path.